torch_xla/csrc/convolution.cpp

```diff
@@ -316,6 +318,548 @@ ConvGrads BuildTransposedConvolutionBackward(
 
 }  // namespace
 
+// Convert a PTXLATensorFormat into string. And this code copied/inspired from
+// TF:
+// https://github.com/tensorflow/tensorflow/blob/7f47eaf439d2b81de1aa24b10ed57eabd519dbdb/tensorflow/core/util/tensor_format.cc#L40
```

Comment:
actually just leave a comment here saying
```
 // -------------Convolution Helper Function Start-------------------------
// Convolution helper functions below are copied from TF2XLA bridge in 
// https://github.com/tensorflow/tensorflow/blob/7f47eaf439d2b81de1aa24b10ed57eabd519dbdb/tensorflow/core/util/tensor_format.cc#L40
```

```
 // -------------Convolution Helper Function End--------------------------
```

Reply:
oh I realized they can be in different files, regardless you can just list all files once instead of right before every function.

Reply:
Thanks, Jack, updated